### PR TITLE
Add vuid 06667, Validate dynamic color write enable

### LINKED ISSUE #3406: Validation for `VK_DYNAMIC_STATE_COLOR_WRITE_ENABLE_EXT` on draw
**Describe the situation in which you encountered the missing validation**
Issue description: When a draw command is issued, there is no check to ensure that the `COLOR_WRITE_ENABLE_EXT` state has been previously set, if the pipeline requires this state. Presumably, the validity of the state (especially regarding the `attachmentCount`) is also not checked at draw time.

**Valid Usage IDs requested**
Please include the valid usage IDs for the checks you are requesting: There is no VUID for this, but the documentation of `VkDynamicState` says: "`VK_DYNAMIC_STATE_COLOR_WRITE_ENABLE_EXT` specifies that the `pColorWriteEnables` state in VkPipelineColorWriteCreateInfoEXT will be ignored and **must** be set dynamically with vkCmdSetColorWriteEnableEXT before any draw call."

### PR DESCRIPTION
**Pull request by ziga-lunarg**
Closes #3406 
Replaces #3408

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -1194,6 +1194,20 @@ bool CoreChecks::ValidatePipelineDrawtimeState(const LAST_BOUND_STATE &state, co
...
         ValidateStatus(pCB, CBSTATUS_VERTEX_INPUT_SET, "Dynamic vertex input not set for this command buffer", vuid.vertex_input);
+    skip |= ValidateStatus(pCB, CBSTATUS_COLOR_WRITE_ENABLE_SET, "Dynamic color write enable not set for this command buffer",
+                           vuid.color_write_enable);
+    if (IsDynamic(pPipeline, VK_DYNAMIC_STATE_COLOR_WRITE_ENABLE_EXT) && pCB->status & CBSTATUS_COLOR_WRITE_ENABLE_SET) {
+        uint32_t blend_attachment_count =
+            pCB->GetCurrentPipeline(VK_PIPELINE_BIND_POINT_GRAPHICS)->create_info.graphics.pColorBlendState->attachmentCount;
+        if (pCB->dynamicColorWriteEnableAttachmentCount < blend_attachment_count) {
+            skip |=
+                LogError(pCB->commandBuffer(), vuid.color_write_enable,
+                         "Currently bound pipeline was created with VkPipelineColorBlendStateCreateInfo::attachmentCount %" PRIu32
```
Comment (sfricke-samsung) [inline layers/core_validation.cpp:1205]: nit: `%s: ` and `caller` are first argument

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -1194,6 +1194,20 @@ bool CoreChecks::ValidatePipelineDrawtimeState(const LAST_BOUND_STATE &state, co
                            "Dynamic vertex input binding stride not set for this command buffer", vuid.vertex_input_binding_stride);
     skip |=
         ValidateStatus(pCB, CBSTATUS_VERTEX_INPUT_SET, "Dynamic vertex input not set for this command buffer", vuid.vertex_input);
+    skip |= ValidateStatus(pCB, CBSTATUS_COLOR_WRITE_ENABLE_SET, "Dynamic color write enable not set for this command buffer",
+                           vuid.color_write_enable);
+    if (IsDynamic(pPipeline, VK_DYNAMIC_STATE_COLOR_WRITE_ENABLE_EXT) && pCB->status & CBSTATUS_COLOR_WRITE_ENABLE_SET) {
+        uint32_t blend_attachment_count =
+            pCB->GetCurrentPipeline(VK_PIPELINE_BIND_POINT_GRAPHICS)->create_info.graphics.pColorBlendState->attachmentCount;
+        if (pCB->dynamicColorWriteEnableAttachmentCount < blend_attachment_count) {
```
Comment (sfricke-samsung) [inline layers/core_validation.cpp:1202]: `!=` ?

Comment (ziga-lunarg) [reply layers/core_validation.cpp:1202]: You're right, thanks!

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

**Pull request merged**